strawberry/experimental/pydantic/object_type.py

```diff
@@ -301,6 +300,57 @@ def to_pydantic_default(self) -> PydanticModel:
     return wrap
 
 
-input = partial(type, is_input=True)
+def input(
+    model: Type[PydanticModel],
+    *,
+    fields: Optional[List[str]] = None,
+    name: Optional[str] = None,
+    is_interface: bool = False,
+    description: Optional[str] = None,
+    directives: Optional[Sequence[StrawberrySchemaDirective]] = (),
+    all_fields: bool = False,
+    use_pydantic_alias: bool = True,
+) -> Callable[..., Type[StrawberryTypeFromPydantic[PydanticModel]]]:
+    """Convenience decorator for creating an input type from a Pydantic model.
+    Equal to partial(type, is_input=True)
+    See https://github.com/strawberry-graphql/strawberry/issues/1830
+    """
+    return type(
+        model=model,
+        fields=fields,
+        name=name,
+        is_input=True,
+        is_interface=is_interface,
+        description=description,
+        directives=directives,
+        all_fields=all_fields,
+        use_pydantic_alias=use_pydantic_alias,
+    )
```

Comment:
technically you could just leave in ```partial(type, is_input=True)``` and put this in the IF TYPE_CHECKING block. Anyways i made it like this so that its less confusing when users introspect. they may not be familar to IF TYPE_CHECKING